Src/FluentAssertions/Equivalency/SelfReferenceEquivalencyOptions.cs

```diff
@@ -245,7 +245,7 @@ public TSelf IncludingInternalFields()
     /// <remarks>
     /// This does not preclude use of `Including`.
     /// </remarks>
-    public TSelf ExcludingFields()
+    protected TSelf ExcludingFields()
```

Comment:
This will make the functionality unavailable and thus isn't something we want.

Reply:
Unavailable for what? No test failed and class is `abstract`?

Reply:
Just because we haven't invoked it directly from our own unit tests doesn't imply that no-one is using it.

To silence analyzers we could perhaps instead write a test that directly invokes this method.

Reply:
I have meant that, because `SelReferenceEquivalencyOptions` is abstract you have to inherit from it and therefore you can access this. Am I missing something? 🤔

Reply:
The case where the derived class is cast to the abstract class.
[SharpLab](https://sharplab.io/#v2:EYLgtghglgdgNAFxAJwK4wD4CEIGcCmABBIQLyEz4DuhAIvslAG74AmAFAJQDcAsAFAB9MsQB0ANQgAbVPj78BEYLgTIIAYwSEAAgCZCOAgIDeAwucIAHZAHsE+TW0KwtkmUWOEA5vgTdCAL4CQQr8enQMzE4gBnj4JgFAA=)

```cs
Base a = new Derived();
_ = a.Value; // error CS0122: 'Base.Value' is inaccessible due to its protection level

abstract class Base
{
    protected int Value { get; }
}

class Derived : Base
{}
```

Reply:
oh.. I see 👍 

What's also interesting: qodana does not complain about the other `public` methods in this class.
Can this rule be disabled only for this file?

Reply:
We always use the `EquivalencyOptions` in `BeEquivalentTo` and we just forgot to add tests that use this method.